docs/spec/rum_v3_context.json

```diff
@@ -0,0 +1,298 @@
+            "description": "Service related information can be sent per event. Provided information will override the more generic information from metadata, non provided fields will be set according to the metadata information.",
+            "properties": {
+                "a": {
+                    "description": "Name and version of the Elastic APM agent",
+                    "type": ["object", "null"],
+                    "properties": {
+                        "n": {
+                            "description": "Name of the Elastic APM agent, e.g. \"Python\"",
+                            "type": ["string", "null"],
+                            "maxLength": 1024
+                        },
+                        "ve": {
+                            "description": "Version of the Elastic APM agent, e.g.\"1.0.0\"",
+                            "type": ["string", "null"],
+                            "maxLength": 1024
+                        },
+                        "ephemeral_id": {
+                            "description": "Free format ID used for metrics correlation by some agents",
+                            "type": ["string", "null"],
+                            "maxLength": 1024
+                        }
+                    }
+                },
+                "framework": {
+                    "description": "Name and version of the web framework used",
+                    "type": ["object", "null"],
+                    "properties": {
+                        "n": {
+                            "type": ["string", "null"],
+                            "maxLength": 1024
+                        },
+                        "ve": {
+                            "type": ["string", "null"],
+                            "maxLength": 1024
+                        }
+                    }
+                },
+                "la": {
+                    "description": "Name and version of the programming language used",
+                    "type": ["object", "null"],
+                    "properties": {
+                        "name": {
+                            "type": ["string", "null"],
+                            "maxLength": 1024
+                        },
+                        "version": {
+                            "type": ["string", "null"],
+                            "maxLength": 1024
+                        }
+                    }
+                },
+                "n": {
+                    "description": "Immutable name of the service emitting this event",
+                    "type": ["string", "null"],
+                    "pattern": "^[a-zA-Z0-9 _-]+$",
+                    "maxLength": 1024
+                },
+                "environment": {
+                    "description": "Environment name of the service, e.g. \"production\" or \"staging\"",
+                    "type": ["string", "null"],
+                    "maxLength": 1024
+                },
+                "runtime": {
```

Comment:
these could be prefilled from user agent since that is the runtime for the RUM agent.